### PR TITLE
Completed Ascending/Descending Sort Buttons

### LINKED ISSUE #290: Add arrow for ascending/descending on sort by buttons
When sorting by GPA/Difficulty/etc in the course view page, add arrow icons indicating whether it's sorted ascending or descending

### PR DESCRIPTION
**Pull request by borjainglefernandez**
## Status
- Done but may need some front end work to make it look better

## GitHub Issues addressed (`#IssueNumber`)
- This PR closes #290 

## What I did
- Added both ascending/descending buttons to the subdepartment pages

## Screenshots
- Before
image

- After
Screenshot 2021-02-08 at 6 26 35 PM

## Tests
- A brief explanation of tests done/written or how reviewers can test your work
- 1.) Open up incognito (I found that the static files don't update on normal browsers)
- 2.) Navigate to browse and click on any subdepartment (i.e. Art)
- 3.) Ensure that "Ascending Order" is the default when navigating to a new sort button (i.e. GPA)
- 4.) Ensure that when hitting the same sort button twice, it alternates between "Ascending Order" and "Descending Order"
- 5.) Ensure that the sort works correctly

## Questions/Discussions/Notes
- I followed the ascending/descending orders listed in sortClasses.js but some of them are not actually ascending, should we change this?
- The design may need some work.

## Merging the PR
- Who should merge this PR?
  - [ ] I will merge it myself
  - [ ] The last reviewer to approve can merge it
- Should the commit history be preserved in the base branch, or is it okay to combine all of them into a single commit ("squash-merge")?
  - [ ] preserve the history
  - [ ] squash-merge

### CODE REVIEW

**Comment by julesfll** [comment]
> I found that the static files don't update on normal browsers
You can do ctrl + f5 on Chrome to clear the cache and reload the static files

**Comment by borjainglefernandez** [comment]
> Good work!
> 
> > I found that the static files don't update on normal browsers
> 
> If you do ctrl + f5 on Chrome it will clear the cache and reload the static files

Thanks for the tip!

File: tcf_website/static/department/sortClasses.js
Comment (jackliu612) [inline tcf_website/static/department/sortClasses.js:137]: These two functions seem like they might be better suited in the common sorting file under static/common/sorting.js
Also, as it stands, these functions are not bound to any buttons as far as I can tell? Look under the file department/bind-sort.js for example.

**Review by jackliu612 — CHANGES_REQUESTED** [review]
Overall good work on adding the indicators they definitely make the sorting more clear to the user. As for the UI you can reach out to the marketing team for mockups or advice. I think one potential option would be to have up/down arrows next to the words number, gpa, rating etc. to reduce button clutter. 
See also my comment about the ascendingOrder and descendingOrder functions.

File: tcf_website/static/department/sortClasses.js
Comment (borjainglefernandez) [reply tcf_website/static/department/sortClasses.js:137]: Cool, I'll definitely move those over. I think you missed it but if you scroll to the top you'll see that they are bonded in department/bind-sort.js.

Comment (jackliu612) [reply tcf_website/static/department/sortClasses.js:137]: Oops my bad!

**Comment by alex-shen1** [comment]
Looks like other people have given some reviews so I'll hold off on that, but with regards to design, you should definitely consider asking D&M to make you a mockup! Not sure whether or not we're still using the request form tbh, but you can always just ping over Discord.

**Comment by borjainglefernandez** [comment]
> ## Status
> * Done but may need some front end work to make it look better
> 
> ## GitHub Issues addressed (`#IssueNumber`)
> * This PR closes #290
> 
> ## What I did
> * Added both ascending/descending buttons to the subdepartment pages
> 
> ## Screenshots
> * Before
>   image
> * After
> 
> Screenshot 2021-02-08 at 6 26 35 PM
> 
> ## Tests
> * A brief explanation of tests done/written or how reviewers can test your work
> * 1.) Open up incognito (I found that the static files don't update on normal browsers)
> * 2.) Navigate to browse and click on any subdepartment (i.e. Art)
> * 3.) Ensure that "Ascending Order" is the default when navigating to a new sort button (i.e. GPA)
> * 4.) Ensure that when hitting the same sort button twice, it alternates between "Ascending Order" and "Descending Order"
> * 5.) Ensure that the sort works correctly
> 
> ## Questions/Discussions/Notes
> * I followed the ascending/descending orders listed in sortClasses.js but some of them are not actually ascending, should we change this?
> * The design may need some work.
> 
> ## Merging the PR
> * Who should merge this PR?
>   
>   * [ ]  I will merge it myself
>   * [ ]  The last reviewer to approve can merge it
> * Should the commit history be preserved in the base branch, or is it okay to combine all of them into a single commit ("squash-merge")?
>   
>   * [ ]  preserve the history
>   * [ ]  squash-merge

image
Changed to look like this.

**Comment by quyenxhuynh** [comment]
Not sure how difficult it would be, but I think having the sort up/down on the current button that is clicked would be more in line with what other sites tend to do. One arrow indicating which way it's being sorted and on hover/click it would swap. Something like:

**Comment by borjainglefernandez** [comment]
> Not sure how difficult it would be, but I think having the sort up/down on the current button that is clicked would be more in line with what other sites tend to do. One arrow indicating which way it's being sorted and on hover/click it would swap. Something like: 

That's actually a great idea, I'll look into that.

**Comment by borjainglefernandez** [comment]
image
Now looks like this. I couldn't find a way to get the ^ and the opposite of that in the actual button with text without the font also changing so I went with the arrows but might be worth looking into the ^ again.

**Review by jackliu612 — APPROVED** [review]

File: tcf_website/static/department/sortClasses.js
Comment (jackliu612) [inline tcf_website/static/department/sortClasses.js:99]: Minor nitpick, but not sure if it's better to use the Unicode hex value instead For example, 
```
arrow = " \u0394" # Up arrow (dummy Unicode value replace with the real one)
```
It's more of a personal preference thing so up to you if you want to change. If you do though, make sure to do the same in the template code for consistency

**Review by jackliu612 — APPROVED** [review]
Overall great work. I was a bit late approving the changes I originally requested but it looks good now. Just left a small nitpick comment. I think the arrows in Quyen's mockup does look a bit cleaner, but I checkout your branch and what you have also looks clear. I think you can make a decision on the comment I left and after that it looks good to merge.

File: tcf_website/static/department/sortClasses.js
Comment (borjainglefernandez) [reply tcf_website/static/department/sortClasses.js]: I agree as well. It'll get rid of any inconsistencies in the future if the arrow doesn't show up on certain computers and os systems etc.

**Comment by borjainglefernandez** [comment]
> Overall great work. I was a bit late approving the changes I originally requested but it looks good now. Just left a small nitpick comment. I think the arrows in Quyen's mockup does look a bit cleaner, but I checkout your branch and what you have also looks clear. I think you can make a decision on the comment I left and after that it looks good to merge.

Just fixed the unicode thing and it should be good to go.

**Review by jackliu612 — APPROVED** [review]

**Pull request merged**
